pyext/src/clustering_rmsd.py

```diff
@@ -139,18 +139,25 @@ def percent_ensemble_explained(ctable, total_num_models):
 
 
 def get_clusters(cutoffs_list, distmat_full, all_models, total_num_models,
-                 run1_all_models, run2_all_models, sysname):
+                 run1_all_models, run2_all_models, sysname, cores):
     # Do Clustering on a Grid
     pvals = []
     cvs = []
     percents = []
+
+    def unpacking_wrapper(arg_tuple):
+        return precision_cluster(*arg_tuple)
+
     with open("%s.ChiSquare_Grid_Stats.txt" % sysname, 'w+') as f1:
-        for c in cutoffs_list:
-            cluster_centers, cluster_members = precision_cluster(
-                    distmat_full, total_num_models, c)
+        with Pool(cores) as p:
```

Comment:
I don't think this will work in Python 2 because Pool doesn't support the context manager protocol until 3.4 or so. IMP code still needs to work with both Python 3 and Python 2.

Reply:
I changed it to not using the context manager. Hopefully, this should run on Python 2 also. In case it doesn't I can run it without `Pool` (using `multiprocessing.Process`)